--- a/drivers/usb/serial/usb_wwan.c
+++ b/drivers/usb/serial/usb_wwan.c
@@ -476,6 +476,12 @@
 	usb_fill_bulk_urb(urb, serial->dev,
 			  usb_sndbulkpipe(serial->dev, endpoint) | dir,
 			  buf, len, callback, ctx);
+	
+	if (dir == USB_DIR_OUT) {
+  		struct usb_device_descriptor *desc = &serial->dev->descriptor; 
+		if (desc->idVendor == cpu_to_le16(0x2C7C)) 
+  			urb->transfer_flags |= URB_ZERO_PACKET; 
+	}
 
 	if (intfdata->use_zlp && dir == USB_DIR_OUT)
 		urb->transfer_flags |= URB_ZERO_PACKET;
